src/lib.rs

```diff
@@ -82,36 +82,47 @@ pub enum NodeError {
     #[fail(display = "Can not insert a node after itself")]
     InsertAfterSelf,
 
+    /// Deprecated, no longer used.
     #[fail(display = "First child is already set")]
     FirstChildAlreadySet,
```

Comment:
I think we should remove them. We cannot avoid a breaking change nevertheless. :)

Reply:
Ok. Removed the last commit, which adds that comments.

Reply:
No, I meant if the types are not used at all, we can completely remove them.

Reply:
Sorry, I misread them X-)

I didn't remove the unused variants because I think their existence are not so harmful, and we can wait until there are unavoidable breaking change (and indextree 4.0.0 is released).

Reply:
Some nervous users might test if such no-longer-used-variants is returned, but (in theory) such variants won't be emitted before and after this pull request.

```rust
match error.downcast<NodeError>().unwrap() {
    NodeError::FirstChildAlreadySet => {
        // User may try to do something here, but
        // this shouldn't have happened before and never happen in future.
    },
    ....
}
```

So it should be safe to leave them (and to remove later).